packages/tabs/src/vaadin-tabs.js

```diff
@@ -236,12 +236,102 @@ class Tabs extends ResizeMixin(ElementMixin(ListMixin(ThemableMixin(PolymerEleme
 
   /** @private */
   _scrollForward() {
-    this._scroll(-this.__direction * this._scrollOffset);
+    const forwardButtonWidth = this._getNavigationButtonVisibleWidth('forward-button');
+    const backButtonWidth = this._getNavigationButtonVisibleWidth('back-button');
+    const scrollerElementBoundingClientRect = this._scrollerElement.getBoundingClientRect();
+    let itemToScrollTo;
+    for (let i = this.items.length - 1; i >= 0; i--) {
+      const isItemVisible = this._isItemVisible(
+        this.items[i],
+        true,
+        forwardButtonWidth,
+        backButtonWidth,
+        scrollerElementBoundingClientRect,
+      );
+      if (isItemVisible) {
+        itemToScrollTo = this.items[i];
+        break;
+      }
+    }
```

Comment:
```suggestion
    const itemToScrollTo = this.items.find((item) => {
      return this._isItemVisible(
        item,
        true,
        forwardButtonWidth,
        backButtonWidth,
        scrollerElementBoundingClientRect,
      )
    });
```

Reply:
Note that `this.items.reverse()` would also modify `this.items` array. if that's not intentional, better use `[...this.items].reverse()`

Reply:
Refactored the code to use your suggestion including Tomi's suggestion.